clients/roscpp/src/libros/topic_manager.cpp

```diff
@@ -1004,11 +1004,34 @@ extern std::string console::g_last_error_message;
 
 void TopicManager::pubUpdateCallback(XmlRpc::XmlRpcValue& params, XmlRpc::XmlRpcValue& result)
 {
+  std::stringstream ss;
+  const std::string& topic = params[1];
+  int is_rosout_topic = (strcmp(topic.c_str(), "rosout") == 0);
+  if (!is_rosout_topic)
+  {
+    ss << "publisherUpdate callback: topic=" << topic.c_str();
+    ss << ", pubs=[";
+  }
+
   std::vector<std::string> pubs;
   for (int idx = 0; idx < params[2].size(); idx++)
   {
+    const std::string& puburi = params[2][idx];
+    if (!is_rosout_topic)
+    {
+      ss << puburi.c_str() << ",";
+    }
+
     pubs.push_back(params[2][idx]);
   }
+
+  if (!is_rosout_topic) 
+  {
+    ss << "]";
+    auto p = ss.str();
+    ROS_INFO("%s", p.c_str());
```

Comment:
Right.  The issue we're trying to troubleshoot right now is to see if we can catch a publisherUpdate call causing topics to disconnect and whether those calls are appearing later than expected.  I need it at INFO because I'm never sure when or which robot it will happen on - although admittedly we could probably just turn DEBUG on for all robots that are deployed but that seems like the same as just setting it to INFO.